lib/FlashMem/FlashMem.cpp

```diff
@@ -0,0 +1,191 @@
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Implememtation of FlashMem.h
+ * @author Gabryel Reyes <gabryelrdiaz@gmail.com>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "FlashMem.h"
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+bool Flash::begin()
+{
+    bool isSuccess = true;
+    EEPROM.begin(EEPROM_SIZE);
+    return isSuccess;
+}
+
+bool Flash::importCredentials(String &ssid, String &password)
+{
+    bool isSuccess = false;
+
+    if (areCredentialsStored())
+    {
+        char temp = 0x00;
+
+        for (uint8_t memoryPosition = 0; memoryPosition <= NVM_SSID_MAX_LENGTH; memoryPosition++)
+        {
+            temp = EEPROM.read(NVM_SSID_ADDRESS + memoryPosition);
+
+            if (0x00 == temp)
+            {
+                break;
+            }
+            else
+            {
+                ssid += temp;
+            }
+        }
+
+        if (0x00 != EEPROM.read(NVM_PASSWORD_ADDRESS))
+        {
+            for (uint8_t memoryPosition = 0; memoryPosition <= NVM_PASSWORD_MAX_LENGTH; memoryPosition++)
+            {
+                temp = EEPROM.read(NVM_PASSWORD_ADDRESS + memoryPosition);
+
+                if (0x00 == temp)
+                {
+                    break;
+                }
+                else
+                {
+                    password += temp;
+                }
+            }
+        }
+
+        Serial.println("\n SSID: " + ssid + "\n Password: " + password);
+
+        isSuccess = true;
+    }
+
+    return isSuccess;
+}
+
+bool Flash::saveCredentials(const String &ssid, const String &password)
+{
+    bool isSuccess = false;
+
+    Flash::clearEEPROM();
```

Comment:
Why is this necessary? You write SSID and password, set the magic numbers if they are not available and finished.